ultralytics/nn/tasks.py

```diff
@@ -655,6 +673,14 @@ def loss(self, batch, preds=None):
         return self.criterion(preds, batch)
 
 
+# NOTE: keep YOLOv10DetectionModel for compatibility with yolov10 pretrained weights.
+class YOLOv10DetectionModel(DetectionModel):
+    """YOLOv10 Detection model."""
+
+    def __init__(self, cfg="yolov10n.yaml", ch=3, nc=None, verbose=True, end2end=True):
+        super().__init__(cfg, ch, nc, verbose, end2end)
```

Comment:
@Laughing-q do we need his class here?

Reply:
intend to eliminate this after this PR merged. https://github.com/ultralytics/ultralytics/pull/13819

Reply:
@Laughing-q merged #13819! I'll remove and test now.

Reply:
@Laughing-q ok I've deleted the class and passed it into safe_load in https://github.com/ultralytics/ultralytics/pull/13113/commits/414f548f9eea62ecec6a765a075f3b73abd5577a.